pkg/services/object/search/service.go

```diff
@@ -26,6 +27,21 @@ type searchClient interface {
 	searchObjects(*execCtx, client.NodeInfo) ([]oid.ID, error)
 }
 
+// Node represents NeoFS storage node that is served by [Service].
+type Node interface {
+	// ForEachContainerNode iterates over all nodes matching the referenced
+	// container's storage policy for now and passes their descriptors into f.
+	// Elements may be repeated.
+	//
+	// Returns [apistatus.ErrContainerNotFound] if referenced container was not
+	// found.
+	ForEachContainerNode(cnr cid.ID, f func(netmapsdk.NodeInfo)) error
+
+	// IsOwnPublicKey checks whether given pubKey assigned to Node in the NeoFS
+	// network map for now.
+	IsOwnPublicKey(pubKey []byte) bool
```

Comment:
I'd just pass the key to `New()`. It can't be changed in runtime and search service needs it for operation --- pass during initialization and then handle it locally. It'd be easier than this API.

Reply:
i went the other way and made an iterator over remote nodes only. Now the service does not need access to the key at all